lib/workers/repository/index.ts

```diff
@@ -54,7 +54,10 @@ export async function renovateRepository(
         return recursiveRes;
       }
       logger.debug(`Automerged but already retried once`);
-    } else {
+    } else if (
+      GlobalConfig.get('dryRun') !== 'lookup' &&
+      GlobalConfig.get('dryRun') !== 'extract'
+    ) {
```

Comment:
Does this function already have its own dry run check? I'm not sure this code is necessary

Reply:
I added this condition because we don't need to access the `ensureDependencyDashboard(config, branches)` when running with extract or lookup mode.

Reply:
you are right this is not necessary because in case we are running with extract or lookup mode the flow should be ended before this point.
I will delete this "if"